snorkel/labeling/model/tuner.py

```diff
@@ -0,0 +1,157 @@
+
+import numpy as np
+import ray
+from ray.tune import Experiment, Trainable
+
+from snorkel.labeling import LabelModel
+
+
+class LabelModelTuner(object):
+    """A hyperparameter tuner for the LabelModel using Ray Tune (https://ray.readthedocs.io/en/latest/tune.html).
+
+    Example
+    -------
+    >>> L_train = np.array([[1, 1, 1, 0], [0, 1, 1, 0], [1, 0, 0, 1], [0, 1, 0, 1]])
+    >>> L_dev = np.array([[1, 1, 1, 0], [0, 1, 1, 0], [1, 0, 0, 1], [0, 1, 0, 1]])
+    >>> Y_dev = np.array([1, 1, 0, 1])
+    >>> label_model = LabelModel(cardinality=2)
+    >>> label_model_tuner = LabelModelTuner(label_model, L_train, L_dev, Y_dev)
+    >>> label_model_tuner = LabelModelTuner(label_model, L_train, L_dev, Y_dev, fit_config={"class_balance" : np.array([0.5, 0.5])})
+
+    Parameters
+    ----------
+    label_model
+        LabelModel instance
+    L_train
+        A matrix with values in {-1,0,1,...,k-1}
+    L_dev
+        A matrix with values in {-1,0,1,...,k-1}
+    Y_dev
+        Gold labels for dev set for tuning hyperparameters
+    metric
+        Metric to maximize while hyperparameter tuning
+    fit_config
+        Arguments for changing LabelModel fit() config defaults
+    """
+
+    def __init__(
+        self,
+        label_model,
+        L_train: np.ndarray,
+        L_dev: np.ndarray,
+        Y_dev: np.ndarray,
+        metric: str = "accuracy",
+        fit_config: Optional[Dict[str, Any]] = {},
+    ) -> None:
+
+        self.label_model = label_model
+        self.L_train = L_train
+        self.L_dev = L_dev
+        self.Y_dev = Y_dev
+        self.metric = metric
+        self.fit_config = fit_config
+
+    def _trainable(self):  # type: ignore
+        """Return a Trainable class definition for Ray tuner with some user-defined parameters from LabelModelTuner class."""
+
+        class LabelModelTrainable(Trainable):
+            """Trainable class for LabelModel for Ray tuner."""
+
+            def _setup(self, config: Optional[Dict[str, Any]]):
+                self.tuner = LabelModelTrainable.tuner  # type: ignore
+                self.tuner.config = config
+                self.timestep = 0
+
+            def _train(self):
+                self.timestep += 1
+                self.tuner.label_model.fit(
+                    self.tuner.L_train, **self.tuner.fit_config, **self.config
+                )
+                score = self.tuner.label_model.score(
+                    self.tuner.L_dev, self.tuner.Y_dev, metrics=[self.tuner.metric]
+                )
+                return score
+
+            def _save(self, checkpoint_dir: str):
+                path = os.path.join(checkpoint_dir, "checkpoint")
+                with open(path, "w") as f:
+                    f.write(json.dumps({"timestep": self.timestep}))
+                return path
+
+            def _restore(self, checkpoint_path: str):
+                with open(checkpoint_path) as f:
+                    self.timestep = json.loads(f.read())["timestep"]
```

Comment:
`json.load(f)`

Reply:
what about the config?

Reply:
This is from the hyberband example [here](https://github.com/ray-project/ray/blob/9ce6dd9b883594cec4ce42bd55962d53088afabe/python/ray/tune/examples/hyperband_example.py#L38) 

On second read, the _setup function there only has timestep while I have others, will fix here